packages/boxel-ui/addon/src/components/drag-and-drop/index.gts

```diff
@@ -0,0 +1,333 @@
+import { fn, hash } from '@ember/helper';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import type { ComponentLike } from '@glint/template';
+
+const isFastBoot = typeof (globalThis as any).FastBoot !== 'undefined';
+
+//Dnd Kanban Board
+export interface ColumnHeaderArgs {
+  title?: string;
+}
+
+export interface DndKanbanBoardArgs<Column> {
+  columnHeader?: ComponentLike<ColumnHeaderArgs>;
+  columns: Column[];
+}
+
+export interface DndKanbanBoardSignature<Column> {
+  Args: DndKanbanBoardArgs<Column>;
+  Blocks: {
+    default: [card: Card, column: Column];
+  };
+}
+
+export class Card {
+  [key: string]: any;
+
+  constructor(...args: any[]) {
+    if (args.length === 1 && typeof args[0] === 'object') {
+      Object.assign(this, args[0]);
+    }
+  }
+}
+
+export class Column {
+  @tracked title: string;
+  @tracked cards: Card[];
+
+  constructor(title: string, cards: Card[] = []) {
+    this.title = title;
+    this.cards = cards;
+  }
+}
+
+export default class DndKanbanBoard extends Component<
+  DndKanbanBoardSignature<Column>
+> {
+  @tracked isModifiersLoaded = false;
+  @tracked DndDraggableItemModifier: any = null;
+  @tracked DndDropTargetModifier: any = null;
+  @tracked DndSortableItemModifier: any = null;
+  @tracked insertAfter: any;
+  @tracked insertAt: any;
+  @tracked insertBefore: any;
+  @tracked removeItem: any;
+  @tracked columns: Column[] = this.args.columns;
+  @tracked hoveredColumnIndex: number | null = null;
+
+  constructor(owner: unknown, args: DndKanbanBoardArgs<Column>) {
+    super(owner, args);
+
+    if (!isFastBoot) {
+      this.loadModifiers();
+    }
+  }
+
+  async loadModifiers() {
+    try {
+      // @ts-expect-error Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'
+      const DndDraggableItemModifier = await import(
+        'ember-draggable-modifiers/modifiers/draggable-item'
+      );
+
+      // @ts-expect-error Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'
+      const DndDropTargetModifier = await import(
+        'ember-draggable-modifiers/modifiers/drop-target'
+      );
+
+      // @ts-expect-error Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'
+      const DndSortableItemModifier = await import(
+        'ember-draggable-modifiers/modifiers/sortable-item'
+      );
+      // @ts-expect-error Dynamic imports are only supported when the '--module' flag is set to 'es2020', 'es2022', 'esnext', 'commonjs', 'amd', 'system', 'umd', 'node16', or 'nodenext'
+      const arrayUtils = await import('ember-draggable-modifiers/utils/array');
```

Comment:
i believe we are using `es2022` everywhere else. is there a reason not to here (@lukemelia @ef4 )?

Reply:
probably worth adding a comment that we are lazily loading these modifiers so they play nice in fastboot

Reply:
I don't think so, but I'd defer to @ef4.